pkg/services/repo_service.go

```diff
@@ -0,0 +1,69 @@
+package services
+
+import (
+	"context"
+	"github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/reposerver/apiclient"
+	"github.com/argoproj/argo-cd/util/db"
+	"github.com/argoproj/argo-cd/util/settings"
+	"github.com/argoproj/gitops-engine/pkg/utils/io"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+)
+
+type RepositoryDB interface {
```

Comment:
Could you add some Godoc for the types and functions in this module?

I'm curious what the idea was here, it looks like the goal was to use the ArgoDB, but hide most of it's functionality?